integration/yarn/src/main/java/tachyon/yarn/ApplicationMaster.java

```diff
@@ -178,8 +179,12 @@ public void requestContainers() throws Exception {
     String[] nodes = {mMasterAddress};
 
     // Make container request for Tachyon master to ResourceManager
+    boolean relaxLocality = true;
+    if (StringUtils.isNotEmpty(mMasterAddress) && !mMasterAddress.equals("localhost")) {
+      relaxLocality = false;
+    }
```

Comment:
Would `relaxLocality = mMasterAddress.isEmpty() || mMasterAddress.equals("localhost")` be sufficient? (We should add some preconditions to prevent `mMasterAddress == null` in the constructor, but that can be a separate PR)